### DISCUSSION TITLE
Partition Wizard error, blank screen on Tasmota 12.5.0

### DISCUSSION THREAD

**Discussion by Gk2E7Aaojf3KDCQ** [arendst/Tasmota#20307] [Q&A]
I have a Shelly Plus Plug US I've migrated to Tasmota 12.5.0. I followed the mgos32-to-tasmota32 to the letter, but run into a big problem on step 8, where it tries to replace the bootloader using Tasmota Safeboot. The Partition Wizard is just a blank screen when it loads.

Error output from the Console after trying to load the partition wizard for Tasmota 12.5.0:

> 1:41:43.454 BRY: Exception> 'type_error' - unsupported operand type(s) for <=: 'nil' and 'int'
> 01:41:43.459 stack traceback:
> 01:41:43.460 : in function factory_migrate_eligible
> 01:41:43.462 : in function show_migrate_to_factory
> 01:41:43.463 : in function page_part_mgr
> 01:41:43.464 : in function <lambda>

I also attempted to use the Berry console to just list the partitions.

`import partition_core var p=partition_core.Partition() print(p)`

Results:

> <instance: Partition([
> ],
> <instance: Partition_otadata(ota_active:ota_0, ota_seq=[,], ota_max=1)>
> )>

From what I've read (#19050 , there should be more partitions. I tried the suggestion in #19050 to use an older version of the partition wizard, as it appears I still have the two app partitions (app_0 and app_1). It always boots into app_1 as that always has the asterisk.

I dug into the partition wizard code a little, to see where the line

`if p.ota_max() <= 0 ` 

return false end" is. It checks for ota_max <= 0, which (I think) is where the error starts. The type errror suggests p.ota_max is returning nil as a result. And the whole thing stalls from there.

Anyone with more knowledge of the partition wizard code care to comment? I've scoured this and the mgos32-to-tasmota projects for clues, but I'm stuck. I should mention the Shelly Plus Plug US has no autoconfig option. People have been using Plug US (no plus) as a substitute, especially since Shelly mislabeled several Plus Plug US devices as just Plug US. I'm wondering if using the Plug US version to flash tasmota is causing problems in the different device. Maybe the partition tables are off? I doubt it, both seem to have 4MB flash and should run fine with the same bootloader.bin, tasmota32solo1.bin, etc.

**Comment by sfromis** [comment]
Step 8? Did you mean step 10 instead?

Since 3 days ago, there is actually a conversion setup for the Shelly Plus Plug US:
https://github.com/tasmota/mgos32-to-tasmota32/releases/download/v13.3.0/mgos32-to-tasmota32-PlusPlugUS.zip

Anyway, I'm not familiar enough with the process to suggest what to do.

**Comment by Gk2E7Aaojf3KDCQ** [comment]
You are right. Step 10 is where I encountered the problem. I reported the issue there as well, the project maintainer said there was an error in step 8 and the bootloader was not replaced. This seems to be a problem with the autoconfiguration. There is no template for the Plus Plug US, although there's a little headway on that front. Some have gotten the device to work with the Plug US .zip, but I've had no success. Question is, why?

I tried that exact .zip file. I kept running into the "app name mismatch" problem. Others had the same problem. There's a "Plug US" .zip that is basically the same exact thing, but with the app name as "Plug US." It seems Shelly misnamed their device.

I thought since some Shelly related topics showed up in this project, I might get some traction on this problem if I asked here. I think Partition Wizard has a bug, it's not reading the partitions right and stalling at the bootloader replacement step. It replaced the stock firmware with Tasmota, and this will brick the device if I try to do a firmware upgrade without replacing the bootloader.

**Comment by Jason2866** [comment]
The issue(s) about the PlugUS are still open in Repo mgos32toTasmota. Since it is not reproducable with
EU devices, no investigation is possible. Needs investigations from an user which does have the device.

**Reply by SevenZero0ne** [reply]
is there any testing i can do? with these fresh "shelly plus plug us" i have before i end up opening them and flashing via serial
